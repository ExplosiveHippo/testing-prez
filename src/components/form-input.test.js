import React from 'react';
import { shallow, mount } from 'enzyme';
import FormInput from "./form-input";

const formInput = shallow(<FormInput />);

const red = "red";

describe("<FormInput />", () => {

  describe('Initalizing <FormInput />', () => {
     it("renders an input element", () => {
       expect(formInput).toMatchSnapshot();
     });

     it("should initialize the state with empty string", () => {
       expect(formInput.state().value).toEqual("");
     });
  });

  describe('On blur of <FormInput /> with empty field', () => {
    beforeEach(() => {
      formInput.find("#input").simulate("change", { target: { value: "" } });
      formInput.find("#input").simulate("blur");
    });

     it("should have error class when field is empty on blur", () => {
       expect(formInput.find("#input").hasClass("error")).toEqual(true);
     });

     it("should have error message when field is empty on blur", () => {
       expect(formInput.find(".error-message").exists()).toEqual(true);
     });
  });
 
  it('should update state value on keyPress', () => {
    formInput.find("#input").simulate("change", { target: { value: "f" } });
    expect(formInput.state().value).toEqual("f");
  });

  it('should have error class when invalid email address is entered on blur', () => {
    formInput.find("#email").simulate("change", { target: { value: "lance@asd" } });
    formInput.find("#email").simulate("blur");
    expect(formInput.find("#email").hasClass("error")).toEqual(true);
  });
});