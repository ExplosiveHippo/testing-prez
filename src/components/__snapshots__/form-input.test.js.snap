// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormInput /> Initalizing <FormInput /> renders an input element 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FormInput />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <Input
                    className=""
                    id="input"
                    onChange={[Function]}
                    type="text"
                    value=""
          />
</div>,
        <div>
          <Input
                    className=""
                    id="email"
                    onChange={[Function]}
                    type="text"
                    value={undefined}
          />
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Input
              className=""
              id="input"
              onChange={[Function]}
              type="text"
              value=""
/>,
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "",
              "id": "input",
              "onChange": [Function],
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          undefined,
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Input
              className=""
              id="email"
              onChange={[Function]}
              type="text"
              value={undefined}
/>,
            undefined,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "className": "",
              "id": "email",
              "onChange": [Function],
              "type": "text",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          undefined,
        ],
        "type": "div",
      },
    ],
    "type": "section",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <Input
                        className=""
                        id="input"
                        onChange={[Function]}
                        type="text"
                        value=""
            />
</div>,
          <div>
            <Input
                        className=""
                        id="email"
                        onChange={[Function]}
                        type="text"
                        value={undefined}
            />
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Input
                className=""
                id="input"
                onChange={[Function]}
                type="text"
                value=""
/>,
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "",
                "id": "input",
                "onChange": [Function],
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            undefined,
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <Input
                className=""
                id="email"
                onChange={[Function]}
                type="text"
                value={undefined}
/>,
              undefined,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "className": "",
                "id": "email",
                "onChange": [Function],
                "type": "text",
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            undefined,
          ],
          "type": "div",
        },
      ],
      "type": "section",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
